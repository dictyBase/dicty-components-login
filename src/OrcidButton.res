external styleToString: ReactDOM.Style.t => string = "%identity"

module Styles = %makeStyles(
  theme => {
    button: ReactDOM.Style.make(
      ~paddingTop=`${theme.spacing(2)->string_of_int}px`,
      ~paddingBottom=`${theme.spacing(2)->string_of_int}px`,
      ~paddingLeft=`${theme.spacing(3)->string_of_int}px`,
      ~paddingRight=`${theme.spacing(3)->string_of_int}px`,
      ~textTransform=`none`,
      ~backgroundColor="#a6ce39",
      (),
    )->ReactDOM.Style.unsafeAddProp(
      "&:hover",
      ReactDOM.Style.make(~backgroundColor="#739d00", ())->styleToString,
    ),
    subtitle: ReactDOM.Style.make(~width="100%", ~color="#4D4D4D", ~fontWeight="600", ()),
  }
)

@genType @react.component
let make = (~text: string, ~handleClick: ReactEvent.Mouse.t => unit) => {
  let classes = Styles.useStyles()

  <MaterialUi_Button
    variant=#Contained
    size=#Large
    fullWidth={true}
    startIcon={<OrcidIcon />}
    className={classes.button}
    onClick={handleClick}>
    <MaterialUi_Typography
      className={classes.subtitle} variant=#Inherit align=#Center color=#TextSecondary>
      {text}
    </MaterialUi_Typography>
  </MaterialUi_Button>
}

let default = make
