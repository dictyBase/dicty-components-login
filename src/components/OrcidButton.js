// @flow
import React from "react"
import { Flex, Box } from "rebass"
import { FontAwesomeIconContainer, OrcidButtonStyle } from "../styles/Buttons"

/**
 * This is the ORCID social button. Since there is not an appropriate FontAwesome icon for this organization, we are currently using a nonbreaking space instead to keep the styling identical.
 */

const OrcidButton = () => {
  return (
    <Flex justify="center">
      <Box w={"90%"} mb={"5px"}>
        <OrcidButtonStyle size="large" variant="raised">
          <FontAwesomeIconContainer>&nbsp;</FontAwesomeIconContainer>
          &nbsp; Sign in with ORCID
        </OrcidButtonStyle>
      </Box>
    </Flex>
  )
}

export default OrcidButton
