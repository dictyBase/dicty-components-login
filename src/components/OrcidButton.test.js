import React from "react"
import renderer from "react-test-renderer"
import OrcidButton from "./OrcidButton"
import "jest-styled-components"

test("matching a snapshot of OrcidButton", () => {
  const component = renderer.create(<OrcidButton />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
