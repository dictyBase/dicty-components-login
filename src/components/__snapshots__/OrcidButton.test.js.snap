// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of OrcidButton 1`] = `
.c1 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  width: 90%;
  margin-bottom: 5px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  border-right: 1px solid black;
  line-height: 45px;
  width: 45px;
  font-size: 1.8em;
}

.c3 {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0;
  font-size: 14px;
  font-weight: normal;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 4px;
  position: relative;
  padding-left: 21px;
  text-align: left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-top: 5px;
  color: #fff;
  border-color: rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
  background-color: #a6ce39;
}

.c3.block {
  display: block;
  width: 100%;
}

.c3.large {
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
}

.c3.xs {
  padding: 1px 5px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 3px;
  width: 20%;
}

.c3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:focus,
.c3:hover,
.c3:active {
  color: #fff;
  background-color: #88aa2a;
  border-color: rgba(0,0,0,0.2);
}

<div
  className="c0 c1"
  is={null}
>
  <div
    className="c2"
    is={null}
  >
    <a
      className="c3"
    >
      <i
        className="c4"
      >
         
      </i>
        Sign in with ORCID
    </a>
  </div>
</div>
`;
